### PR TITLE
Feature/basic structure

### LINKED ISSUE #2: Template project for aquality-selenium-dotnet

### PR DESCRIPTION
**Pull request by paveliam**
Add basic test

### CODE REVIEW

File: Aquality.Selenium.Template/Aquality.Selenium.Template.sln [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+Global
+	GlobalSection(SolutionConfigurationPlatforms) = preSolution
+		Debug|Any CPU = Debug|Any CPU
+		Release|Any CPU = Release|Any CPU
+	EndGlobalSection
+	GlobalSection(ProjectConfigurationPlatforms) = postSolution
+		{21B3F61F-E039-492C-A959-73049136EB4A}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
+		{21B3F61F-E039-492C-A959-73049136EB4A}.Debug|Any CPU.Build.0 = Debug|Any CPU
+		{21B3F61F-E039-492C-A959-73049136EB4A}.Release|Any CPU.ActiveCfg = Release|Any CPU
+		{21B3F61F-E039-492C-A959-73049136EB4A}.Release|Any CPU.Build.0 = Release|Any CPU
```
Comment (mialeska) [inline Aquality.Selenium.Template/Aquality.Selenium.Template.sln:19]: wonder do we really need both "Debug" and "Release" configurations for all projects in autotest solution

File: Aquality.Selenium.Template/Aquality.Selenium.Template/Forms/TopBarMenu.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
+﻿using Aquality.Selenium.Forms;
+using OpenQA.Selenium;
+using System.Collections.Generic;
+
+namespace Aquality.Selenium.Template.Forms
+{
+    public class TopBarMenu : Form
+    {
+        private readonly IDictionary<Item, string> menuItems = new Dictionary<Item, string>
```
Comment (mialeska) [inline Aquality.Selenium.Template/Aquality.Selenium.Template/Forms/TopBarMenu.cs]: should it be static?

File: Aquality.Selenium.Template/Aquality.Selenium.Template/Forms/TopBarMenu.cs [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        {
+            { Item.ContactUs, "//div[@id='primary-navigation']//li[contains(@class, 'contact-us menu')]//a" }
+        };
+
+        public TopBarMenu() : base(By.Id("masthead"), "Header")
+        {
+        }
+
+        public void OpenHeaderMenu(Item menuItem)
```
Comment (mialeska) [inline Aquality.Selenium.Template/Aquality.Selenium.Template/Forms/TopBarMenu.cs]: should we use short or full form for single-line methods?

**Review by mialeska — APPROVED** [review]

File: Aquality.Selenium.Template/Aquality.Selenium.Template.sln (hunk 1 above)
Comment (paveliam) [reply Aquality.Selenium.Template/Aquality.Selenium.Template.sln:19]: fixed

File: Aquality.Selenium.Template/Aquality.Selenium.Template/Forms/TopBarMenu.cs (hunk 3 above)
Comment (paveliam) [reply Aquality.Selenium.Template/Aquality.Selenium.Template/Forms/TopBarMenu.cs]: Yes, we can user short form

File: Aquality.Selenium.Template/Aquality.Selenium.Template/Forms/TopBarMenu.cs (hunk 2 above)
Comment (paveliam) [reply Aquality.Selenium.Template/Aquality.Selenium.Template/Forms/TopBarMenu.cs]: yes, should be static

File: Aquality.Selenium.Template/Aquality.Selenium.Template.SpecFlow/Aquality.Selenium.Template.SpecFlow.csproj [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  <PropertyGroup>
+    <TargetFramework>netcoreapp3.0</TargetFramework>
+
+    <IsPackable>false</IsPackable>
+
+    <Configurations>Release</Configurations>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Aquality.Selenium" Version="2.3.1" />
```
Comment (DmitryBogatko) [inline Aquality.Selenium.Template/Aquality.Selenium.Template.SpecFlow/Aquality.Selenium.Template.SpecFlow.csproj:12]: do not we want to use the latest here?

File: Aquality.Selenium.Template/Aquality.Selenium.Template.SpecFlow/Hooks/BrowserHooks.cs [hunk 5]
```diff
@@ -0,0 +1,15 @@
+﻿using Aquality.Selenium.Browsers;
+using TechTalk.SpecFlow;
+
+namespace Aquality.Selenium.Template.Glue.Hooks
```
Comment (DmitryBogatko) [inline Aquality.Selenium.Template/Aquality.Selenium.Template.SpecFlow/Hooks/BrowserHooks.cs]: 'glue' not so readable as 'steps' or 'stepdefinitions'
perhaps better to use 'steps'?

File: Aquality.Selenium.Template/Aquality.Selenium.Template.SpecFlow/StepDefinitions/DemoSteps.cs [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+            contactUsPage.SetName(contactUsInfo.Name)
+                .SetCompany(contactUsInfo.Comment)
+                .SetPhone(contactUsInfo.Phone)
+                .SetComment(contactUsInfo.Comment);
+        }
+        
+        [When(@"I accept Privacy and Cookies Policy")]
+        public void WhenIAcceptPrivacyAndCookiesPolicy()
+        {
+            var contactUsPage = new ContactUsPage();
```
Comment (DmitryBogatko) [inline Aquality.Selenium.Template/Aquality.Selenium.Template.SpecFlow/StepDefinitions/DemoSteps.cs]: can't we re-use 'contactUsPage' field?

File: Aquality.Selenium.Template/Aquality.Selenium.Template.SpecFlow/StepDefinitions/DemoSteps.cs [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+        [Given(@"Main page is opened")]
+        public void GivenMainPageIsOpened()
+        {
+            AqualityServices.Browser.GoTo(Configuration.Configuration.StartUrl);
+        }
+        
+        [When(@"I open Contact us page")]
+        public void WhenIOpenContactUsPage()
+        {
+            var mainPage = new MainPage();
```
Comment (DmitryBogatko) [inline Aquality.Selenium.Template/Aquality.Selenium.Template.SpecFlow/StepDefinitions/DemoSteps.cs]: maybe better to move pages initializing in the constructor?

File: Aquality.Selenium.Template/Aquality.Selenium.Template/Configuration/Environment.cs [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+
+namespace Aquality.Selenium.Template.Configuration
+{
+    internal static class Environment
+    {
+        public static ISettingsFile CurrentEnvironment
+        {
+            get
+            {
+                var envName = EnvironmentConfiguration.GetVariable("environment") ?? "Stage";
```
Comment (DmitryBogatko) [inline Aquality.Selenium.Template/Aquality.Selenium.Template/Configuration/Environment.cs]: do not we want to move default value into settings.json?

**Review by DmitryBogatko — CHANGES_REQUESTED** [review]

File: Aquality.Selenium.Template/Aquality.Selenium.Template.SpecFlow/Aquality.Selenium.Template.SpecFlow.csproj (hunk 4 above)
Comment (paveliam) [reply Aquality.Selenium.Template/Aquality.Selenium.Template.SpecFlow/Aquality.Selenium.Template.SpecFlow.csproj:12]: nuget does not allow it

File: Aquality.Selenium.Template/Aquality.Selenium.Template.SpecFlow/Hooks/BrowserHooks.cs (hunk 5 above)
Comment (paveliam) [reply Aquality.Selenium.Template/Aquality.Selenium.Template.SpecFlow/Hooks/BrowserHooks.cs]: By mistake, copied from another solution

File: Aquality.Selenium.Template/Aquality.Selenium.Template/Configuration/Environment.cs (hunk 8 above)
Comment (pavelanihimovsky) [reply Aquality.Selenium.Template/Aquality.Selenium.Template/Configuration/Environment.cs:13]: EnvironmentConfiguration.GetVariable doesn't read value from settings file

**Review by knysh — APPROVED** [review]

**Pull request merged**